src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -348,20 +433,88 @@ private async Task<IAnonymizer> CreateAnonymizerAsync(CancellationToken cancella
             EnsureArg.IsNotNull(progress, nameof(progress));
             EnsureArg.IsNotNull(sharedQueryParametersList, nameof(sharedQueryParametersList));
 
-            // Current batch will be used to organize a set of search results into a group so that they can be committed together.
-            string currentBatchId = batchIdPrefix + "-" + progress.Page.ToString("d6");
-
             List<Tuple<string, string>> queryParametersList = new List<Tuple<string, string>>(sharedQueryParametersList);
             if (progress.ContinuationToken != null)
             {
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.ContinuationToken, progress.ContinuationToken));
             }
 
-            if (!string.IsNullOrEmpty(_exportJobRecord.ResourceType))
+            var requestedResourceTypes = _exportJobRecord.ResourceType?.Split(',');
+            var filteredResources = new HashSet<string>();
+            if (_exportJobRecord.Filters != null)
+            {
+                foreach (var filter in _exportJobRecord.Filters)
+                {
+                    filteredResources.Add(filter.ResourceType.ToUpper(CultureInfo.InvariantCulture));
+                }
+            }
+
+            if (progress.CurrentFilter != null)
             {
-                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, _exportJobRecord.ResourceType));
+                await ProcessFilterForCompartment(exportJobConfiguration, progress, queryParametersList, batchIdPrefix + "-filter", cancellationToken);
             }
 
+            if (_exportJobRecord.Filters != null && _exportJobRecord.Filters.Any(filter => !progress.CompletedFilters.Contains(filter)))
```

Comment:
>&& _exportJobRecord.Filters.Any(filter => !progress.CompletedFilters.Contains(filter)) [](start = 49, length = 86)

nit: It seems like we are checking for the same condition in the for-each loop below. Can we remove this check (since functionality will not change) in order to avoid looping through the filters twice?